### ISSUE TITLE
replaceState is overridden by useForm's remember

### ISSUE THREAD

**Issue by mattkingshott** [inertiajs/inertia#585]
I have a tab component on a page that updates the query string `tab` key when switching e.g. `?tab=posts`.

The following code is used to do that:

```javascript
let query = new URLSearchParams(document.location.search.substring(1));

query.set('tab', 'post');

query = query.toString();

query = query === '' ? '' : `?${query}`;

let url = new URL(window.location.href);
url.search = query;

let state = history.state;
state.url = `${url.pathname}${query}`;

window.history.replaceState(state, '', url.href);
```

Unfortunately, `replaceState` is being immediately overridden by Inertia also calling `replaceState` internally. 

In my particular case, I've tracked it down to `{remember: true}` within `useForm`. If I set remember to `false`, this solves the problem, however I don't want to loose the functionality provided by `remember`.

Is there a way to hook into what Inertia is doing and run my code after it has finished? Alternatively, what would be the correct way of altering the query string so it doesn't interfere with Inertia?

Thanks!

**Comment by reinink** [comment]
Hey there @mattkingshott! This is the same issue #585, so I'm going to close this in favour of that PR. 👍 

TLDR; Let Inertia handle your client-side history state by making regular Inertia visits. This does, however, require a server-side visit.
